sunkit_pyvista/plotter.py

```diff
@@ -584,3 +586,167 @@ def plot_limb(self, m, *, radius=0.02, **kwargs):
         limb_block.add_field_data(color, "color")
         self.add_mesh(limb_block, color=color, **kwargs)
         self._add_mesh_to_dict(block_name="limbs", mesh=limb_block)
+
+
+class CartesianPlotter(pv.Plotter):
+    """
+    A plotter for 3D data in a Cartesian box.
+
+    This class inherits `pyvista.Plotter`. It is used to visualize 3D vector field lines
+    (e.g., magnetic field lines of solar active regions) traced by `streamtracer`.
+
+    Parameters
+    ----------
+    kwargs : dict
+        All other keyword arguments are passed through to `pyvista.Plotter`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def define_vector_field(self, vectors, *args, **kwargs):
+        """
+        Define a 3D vector field.
+
+        Parameters
+        ----------
+        vectors : ndarray
+            A (nx, ny, nz, 3) array representing the 3D vector field.
+        args: list
+            arguments for `streamtracer.VectorGrid`.
+        kwargs: dict
+            keyword arguments for `streamtracer.VectorGrid`.
+        """
+        if kwargs.get("grid_spacing") is None:
+            if kwargs.get("grid_coords") is None:
+                kwargs["grid_spacing"] = [1, 1, 1]
+        self.grid = VectorGrid(vectors.astype(np.float64), *args, **kwargs)
+        self.xcoords = self.grid.xcoords.astype(np.float64)
+        self.ycoords = self.grid.ycoords.astype(np.float64)
+        self.zcoords = self.grid.zcoords.astype(np.float64)
+        self.x, self.y, self.z = np.meshgrid(self.xcoords, self.ycoords, self.zcoords, indexing="ij")
+
+    def _create_mesh(self):
+        """
+        Create a `pyvista.StructuredGrid` mesh from the 3D vector field.
+        """
+        mesh = pv.StructuredGrid(self.x, self.y, self.z)
+        vectors = self.grid.vectors.transpose(2, 1, 0, 3).reshape(-1, 3)
+        mesh["vectors"] = vectors
+        mesh.active_vectors_name = "vectors"
+        magnitudes = np.linalg.norm(vectors, axis=-1)
+        mesh["magnitudes"] = magnitudes
+        mesh.active_scalars_name = "magnitudes"
+        self.mesh = mesh
+
+    def plot_outline(self, color="black", **kwargs):
+        """
+        Plot the outline of the 3D vector field.
+
+        Parameters
+        ----------
+        color : str, optional
+            The color of the outline.
+            Default is 'black'.
+        kwargs : dict
+            Keyword arguments for `pyvista.Plotter.add_mesh`.
+        """
+        self._create_mesh()
+        self.add_mesh(self.mesh.outline(), color=color, **kwargs)
+
+    def plot_boundary(self, boundary="bottom", component=2, cmap="gray", **kwargs):
```

Comment:
```suggestion
    def plot_boundary(self, boundary="bottom", *, component=2, cmap="gray", **kwargs):
```
The later arguments should be keyword only, to avoid calls like `plot_boundary("bottom", 2, 'gray')`, and instead force calls like `plot_boundary("bottom", component=2, cmap='gray')`.